pkg/config/config.go

```diff
@@ -109,7 +111,18 @@ func (cfg *Config) Host(ref string) (remote.CredentialFunc, bool, error) {
 		if !ok {
 			return nil, fmt.Errorf("not found matched hostname %s in config", refURL.Host)
 		}
-
+		// try to finds auth for a given host in docker's config.json settings.
+		if len(auth.Auth) == 0 {
+			config := config.LoadDefaultConfigFile(os.Stderr)
+			authConfig, err := config.GetAuthConfig(refURL.Host)
+			if err != nil {
+				return nil, err
+			}
+			if len(authConfig.Username) == 0 || len(authConfig.Password) == 0 {
+				return nil, fmt.Errorf("no auth from docker config for host %s", refURL.Host)
```

Comment:
Maybe the registry allows an anonymous request?

Reply:
> Maybe the registry allows an anonymous request?

If users use `docker` or `nerdctl` login the registry, can we get the auth? Actually I don't find the auth, only the username and password.

Reply:
Maybe the docker config.json is empty for the registry allowing anonymous requests.